src/js/index.js

```diff
@@ -0,0 +1,191 @@
+
+	// ui update
+	updateMenuListElementView(menuListItem, updatedMenu);
+}
+
+function removeMenuItem(menuListItem, index) {
+	const isConfrimed = confirm("정말 삭제하시겠습니까?");
+
+	if (!isConfrimed) {
+		return;
+	}
+	// data update
+	state.delete(index);
+
+	// ui update
+	// 메뉴 리스트 전체 render를 새로하는게 아닌,
+	// 변경이 발생한 해당 menu list item만 삭제 후
+	// 메뉴 개수 부분만 새로 render
+	menuListItem.remove();
+	renderMenuCount();
+}
+
+function onMenuClicked(e) {
+	const menuListItem = e.target.closest(".menu-list-item");
+	const menuListItemIndex = Number(menuListItem?.dataset.index);
+
+	// event target이 메뉴 <li> 안에 있지 않다면 리턴
+	if (!menuListItem) {
+		return;
+	}
+
+	// 클릭된 메뉴 <li> 가 메뉴 리스트 UI 내부에 존재하지 않으면 리턴
+	if (!menuList.contains(menuListItem)) {
+		return;
+	}
+
+	// 현재 클릭된 event target이 <button>이 아니면 리턴
+	if (e.target.tagName !== "BUTTON") {
+		return;
+	}
+
+	switch (e.target.dataset.action) {
+		case "edit":
+			updateMenuName(menuListItem, menuListItemIndex);
+			break;
+		case "delete":
+			removeMenuItem(menuListItem, menuListItemIndex);
+			break;
+		case "toggle-soldout":
+			toggleMenuSoldOut(menuListItem, menuListItemIndex);
+			break;
+		default:
+			console.error(`Unexpected action: ${e.target.dataset.action}`);
+			return;
+	}
+}
+
+function onMenuTypeClicked(e) {
+	const clickedMenuTypeButton = e.target.closest(".cafe-category-name");
+
+	// event target이 menuTypeSelectButtonWarpper 안에 있지 않다면 리턴
+	if (!clickedMenuTypeButton) {
+		return;
+	}
+
+	// 현재 클릭된 event target이 <button>이 아니면 리턴
+	if (e.target.tagName !== "BUTTON") {
+		return;
+	}
+
+	const clickedMenuType = e.target.dataset.categoryName;
+	// 현재 클릭된 <button>의 "data-category-name" 속성에 할당된 값이 MenuType에 정의된 값이 아니라면 리턴
+	if (!MenuTypeUtil.isMenuType(clickedMenuType)) {
+		return;
+	}
+
+	state.setSelectedMenuType(clickedMenuType);
+	renderMenu();
+}
+
+function getMenuListItemHTMLString(menu, index) {
+	const soldOutStatusClassName = menu.isSoldOut ? "sold-out" : "";
+
+	return `
+<li class="menu-list-item d-flex items-center py-2" data-index=${index}>
+	<span class="w-100 pl-2 menu-name ${soldOutStatusClassName}">${menu.name}</span>
+	<button type="button" class="bg-gray-50 text-gray-500 text-sm mr-1 menu-sold-out-button" data-action="toggle-soldout">
+		품절
+ 	</button>
+	<button type="button" class="bg-gray-50 text-gray-500 text-sm mr-1 menu-edit-button" data-action="edit">
+    	수정
+  	</button>
+  	<button type="button" class="bg-gray-50 text-gray-500 text-sm menu-remove-button" data-action="delete">
+		삭제
+  	</button>
+</li>
+`;
+}
+
+function renderMenu() {
+	renderMenuList();
+	renderMenuCount();
+	renderMenuHeader();
+}
+
+function renderMenuList() {
+	// 기존에 menuList의 innerHTML에 존재하던 <li> 모두 제거
+	while (menuList.firstChild) {
+		menuList.firstChild.remove();
+	}
+
+	// 현재 state의 해당하는 Menu 배열을 불러와, 각각의 menu를 기반으로 <li> Element를 생성하여 menuList에 삽입
+	state.getMenuList().forEach(
+		(menu, index) => menuList.insertAdjacentHTML("beforeend", getMenuListItemHTMLString(menu, index))
```

Comment:
```suggestion
function renderMenuItem(menu,index){
    menuList.insertAdjacentHTML("beforeend", getMenuListItemHTMLString(menu, index));
}

state.getMenuList().forEach(renderMenuItem);
```

메뉴 아이템을 추가하는 부분을 분리하면 좀 더 코드 읽기가 쉬울 것 같아요~!

Reply:
아하 메뉴 리스트 그려주는 부분을

- (사전 작업) + 메뉴 리스트 그려주는 부분
- 메뉴 리스트 아이템 하나 그려주는 부분

이렇게 분리하는 방향을 말씀해주신 거군요!

"함수는 하나의 일만 한다는 원칙"에 잘 부합하는 것 같네요!

감사합니다! :)